pandas/core/generic.py

```diff
@@ -3779,6 +3779,8 @@ def _take(
 
         See the docstring of `take` for full explanation of the parameters.
         """
+        if axis == 0 and np.array_equal(indices, np.arange(0, len(self))):
```

Comment:
Can we do a quick check of how costly this is compared to the actual `take`? (to ensure we don't introduce a performance regression for the other cases)

Maybe could also first check if the length of `indices` is equal to `len(self)`

Reply:
> Maybe could also first check if the length of `indices` is equal to `len(self)`

`np.array_equal` actually already does that, so doing that ourselves is not needed

Reply:
I profiled this, the cost is reduced if the DataFrame gets larger. 

The actual problem is a bit different though: array_equal has no early exit, even if the first value is different, it checks the whole array (wanted to bring this up tomorrow as well). We should probably write something ourselves, because I guess we will need that a couple of times.